tidy3d/components/medium.py

```diff
@@ -322,53 +324,67 @@ def _validate_numiters(cls, val):
 
-    Note
-    ----
-    This frequency-domain equation is implemented in the time domain using complex-valued fields.
+        .. math::
 
-    Note
-    ----
-    Different field components do not interact nonlinearly. For example,
-    when calculating :math:`P_{NL, x}`, we approximate :math:`|E|^2 \\approx |E_x|^2`.
-    This approximation is valid when the E field is predominantly polarized along one
-    of the x, y, or z axes.
+            P_{NL} = P_{TPA} + P_{FCA} + P_{FCPD} \\\\
+            P_{TPA} = -\\frac{4}{3}\\frac{c_0^2 \\varepsilon_0^2 n_0^2 \\beta}{2 i \\omega} |E|^2 E \\\\
+            P_{FCA} = -\\frac{c_0 \\varepsilon_0 n_0 \\sigma N_f}{i \\omega} E \\\\
+            \\frac{dN_f}{dt} = \\frac{8}{3}\\frac{c_0^2 \\varepsilon_0^2 n_0^2 \\beta}{8 q_e \\hbar \\omega} |E|^4 - \\frac{N_f}{\\tau} \\\\
+            N_e = N_h = N_f \\\\
+            P_{FCPD} = \\varepsilon_0 2 n_0 \\Delta n (N_f) E \\\\
+            \\Delta n (N_f) = (c_e N_e^{e_e} + c_h N_h^{e_h})
 
-    Note
-    ----
-    The implementation is described in::
+        In these equations, :math:`n_0` means the real part of the linear
+        refractive index of the medium.
 
-        N. Suzuki, "FDTD Analysis of Two-Photon Absorption and Free-Carrier Absorption in Si
-        High-Index-Contrast Waveguides," J. Light. Technol. 25, 9 (2007).
+        The nonlinear constitutive relation is solved iteratively; it may not converge
+        for strong nonlinearities. Increasing :attr:`tidy3d.NonlinearSpec.num_iters` can
+        help with convergence.
+
+        For complex fields (e.g. when using Bloch boundary conditions), the nonlinearity
+        is applied separately to the real and imaginary parts, so that the above equation
+        holds when both :math:`E` and :math:`P_{NL}` are replaced by their real or imaginary parts.
+        The nonlinearity is only applied to the real-valued fields since they are the
+        physical fields.
+
+        Different field components do not interact nonlinearly. For example,
+        when calculating :math:`P_{NL, x}`, we approximate :math:`|E|^2 \\approx |E_x|^2`.
+        This approximation is valid when the :math:`E` field is predominantly polarized along one
+        of the ``x``, ``y``, or ``z`` axes.
+
+        The implementation is described in::
+
+            N. Suzuki, "FDTD Analysis of Two-Photon Absorption and Free-Carrier Absorption in Si
+            High-Index-Contrast Waveguides," J. Light. Technol. 25, 9 (2007).
+
+        .. TODO add links to notebooks here.
 
     Example
     -------
     >>> tpa_model = TwoPhotonAbsorption(beta=1)
     """
 
+    use_complex_fields: bool = pd.Field(
+        False,
+        title="Use complex fields",
+        description="Whether to use the old deprecated complex-fields implementation. "
+        "The default real-field implementation is more physical and is always "
+        "recommended; this option is only available for backwards-compatibility "
```

Comment:
Since you're removing the documentation of the old method, let's mention here explicitly something like "backwards compatibility with Tidy3D version < 2.8." Then people can go to our docs and pull up the old documentation if they need to. Also in the other places this argument is defined.